src/Evolution/Systems/Cce/InterfaceManagers/GhInterpolationStrategies.hpp

```diff
@@ -0,0 +1,35 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Time/Tags.hpp"
+
+
+namespace Cce {
+namespace InterfaceManagers{
+
+enum class InterpolationStrategy { EveryStep, EverySubstep };
+
+/*!
+ * \brief Determines whether the element should interpolate for the current
+ * state of its `box`, depending on the
+ * `Cce::InterfaceManagers::InterpolationStrategy` used by the associated
+ * `Cce::InterfaceManagers::GhInterfaceManager`.
+ */
+template <typename DbTagList>
+bool should_interpolate_for_strategy(
+    const db::DataBox<DbTagList>& box,
+    const InterpolationStrategy strategy) noexcept {
+  if(strategy == InterpolationStrategy::EverySubstep) {
+    return true;
+  }
+  if(strategy == InterpolationStrategy::EveryStep) {
```

Comment:
This is a little bit difficult, because the interface manager is specified at runtime (by options), so whether we want every step vs every substep (or something else) is also only known at runtime, so the `InterpolationStrategy` has to be a runtime quantity. And, it has to be available to the DG element array, where the interface manager is stored on the cce worldtube component. The solution here was to just have the enum `InterpolationStrategy` in the const global cache.

However, now that you ask, a similarly effective solution would be to make the `InterpolationStrategy` a factory-creatable abstract base class, and each individual version a derived class, then the `std::unique_ptr` can sit in the const global cache, and have a `should_interpolate` function. Do you think that would better?

Reply:
oops, I'm wrong -- this is less direct than I thought because it would need the `should_interpolate` function to be a virtual, but it wouldn't be able to be a template like it is here.

Reply:
It sounds like it’s a hassle, so let’s leave as is. Thanks for the explanation!